packages/massa-web3/src/experimental/basicElements/address.ts

```diff
@@ -139,18 +180,22 @@ export class Address {
    * @returns The serialized address string.
    */
   toString(): string {
-    const versionedBytes = this.versioner.attach(this.version, this.bytes)
-    return `${ADDRESS_PREFIX}${
+    // skip address type bytes
+    const versionedBytes = this.bytes.slice(
+      varint.encodingLength(this.getType())
+    )
+    return `${
       this.isEOA ? ADDRESS_USER_PREFIX : ADDRESS_CONTRACT_PREFIX
     }${this.serializer.serialize(versionedBytes)}`
   }
 
   /**
-   * Get byte length of address in binary format .
+   * Get address in binary format from a bytes buffer.
+   * The address should be the first element in the buffer.
    *
-   * @returns The address length in bytes.
+   * @returns The address in bytes format.
    */
-  static getByteLength(data: Uint8Array): number {
+  static extractFromBuffer(data: Uint8Array): Uint8Array {
```

Comment:
Don't you think that you should also returned the number of bytes used from this buffer to let the caller continue his potential deserailization steps.

Reply:
i think it would make the code more complex.
The lenght is easily found using `res.length`

Reply:
It may be true with the current implementation, but it's not for others (e.g., the array using Args). I've proposed such an interface because it's classic when extracting data from a buffer. Having consistent function signatures is something important, in my humble opinion.